packages/react/src/components/Toolbar/ToolbarItem.tsx

```diff
@@ -147,7 +198,12 @@ class ToolbarItem extends UIComponent<WithAsProp<ToolbarItemProps>, ToolbarItemS
       })
     }
 
-    return renderedItem
+    return (
+      <>
+        <Ref innerRef={this.itemRef}>{renderedItem}</Ref>
+        {submenu}
+      </>
```

Comment:
based on our discussion and [ARIA spec](https://www.w3.org/TR/wai-aria-practices/examples/toolbar/toolbar.html), toolbar button and menu should be wrapped in one div so then a screen reader will work correctly. @jurokapsiar correct me if I am wrong

Another aspect, when focus is in menu and press ESC, menu should get closed and focus should go on the button.
In theory, this type of keyboard event should be catched by wrapper element, which wraps toolbar button and menu. Let me know what you think.

Reply:
Makes sense. I added a wrapper slot, it is rendered only if there is a menu prop defined on the item, no matter whether the menu is open ( = rendered) or closed ( = not in dom).